### PR TITLE
Connect policy events to Azure events

### PR DESCRIPTION
**Pull request by lfu**
Connect VM power on/off policy events to Azure events.
Also add Azure event for instance suspend.

https://bugzilla.redhat.com/show_bug.cgi?id=1418755

cc @gmcculloug

### CODE REVIEW

**Comment by lfu** [comment]
@bronaghs Please review.

File: content/automate/ManageIQ/System/Event/EmsEvent/Azure.class/virtualmachines_poweroff_endrequest.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  attributes:
+    display_name: 
+    name: virtualMachines_powerOff_EndRequest
+    inherits: 
+    description: 
+  fields:
+  - rel4:
+      value: "/System/event_handlers/event_action_refresh?target=ems"
+  - rel5:
+      value: "/System/event_handlers/event_action_policy?target=src_vm&policy_event=vm_suspend&param="
```
Comment (gmcculloug) [inline content/automate/ManageIQ/System/Event/EmsEvent/Azure.class/virtualmachines_poweroff_endrequest.yaml]: @bronaghs @lfu Is this correct?  The `virtualMachines_powerOff` provider event corresponds to a `vm_suspend`.

Comment (lfu) [inline content/automate/ManageIQ/System/Event/EmsEvent/Azure.class/virtualmachines_poweroff_endrequest.yaml]: That is the event received when suspending an instance. 
@bronaghs Please confirm.

Comment (bronaghs) [inline content/automate/ManageIQ/System/Event/EmsEvent/Azure.class/virtualmachines_poweroff_endrequest.yaml]: Hi @lfu @gmcculloug 


Azure does not actually support the instance *suspend* operation, so when the user suspends an Azure instance in ManageIQ, we actually stop it on the provider:
https://github.com/ManageIQ/manageiq-providers-azure/blob/master/app/models/manageiq/providers/azure/cloud_manager/vm/operations/power.rb#L12

This is why you receive a  ```virtualMachines_powerOff``` event when you "suspend" an Azure instance.

I am thinking we should just disable the suspend button and force the user to power it off instead, this would mirror the functionality in Azure.

Comment (gmcculloug) [inline content/automate/ManageIQ/System/Event/EmsEvent/Azure.class/virtualmachines_poweroff_endrequest.yaml]: Then from what I understand, it is ok to process this event but we should be treating it like a power_off and not a suspend.  @bronaghs Sound good?

Comment (bronaghs) [inline content/automate/ManageIQ/System/Event/EmsEvent/Azure.class/virtualmachines_poweroff_endrequest.yaml]: @gmcculloug - yep, sounds good.

Comment (gmcculloug) [inline content/automate/ManageIQ/System/Event/EmsEvent/Azure.class/virtualmachines_poweroff_endrequest.yaml]: @lfu Please change this to raise a `power_off` policy event..

Comment (lfu) [inline content/automate/ManageIQ/System/Event/EmsEvent/Azure.class/virtualmachines_poweroff_endrequest.yaml]: @gmcculloug @bronaghs 
So a ```power_off``` policy event would be raised for this event and the Suspend task from UI should be removed. Is it correct?

Comment (gmcculloug) [inline content/automate/ManageIQ/System/Event/EmsEvent/Azure.class/virtualmachines_poweroff_endrequest.yaml]: Yes, the power_off event should raise a power_off policy event.

The suspend task should be the focus of a separate PR.  I would suggest that @bronaghs open a git issue with the details and assign it to a provider team member.

**Comment by simaishi** [comment]
Euwe backport (to manageiq repo) details:

```
$ git log -1
commit a995bdb7a41e589cac9a5ccd8316729b03ca93f0
Author: Satoe Imaishi <simaishi@redhat.com>
Date:   Mon Mar 13 13:28:27 2017 -0400

    Merge pull request #34 from lfu/azure_policy_event_1418755
    
    Connect policy events to Azure events
    (cherry picked from commit 94827ec59c2594d5948e03ceb79947bfde8081de)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1421158
```
